include/oneapi/tbb/collaborative_call_once.h

```diff
@@ -185,6 +185,8 @@ class collaborative_once_flag : no_copy {
                     // The moonlighting threads are not expected to handle exceptions from user functor.
                     // Therefore, no exception is expected from assist().
                     shared_runner->assist();
+                    // Read m_state while holding the guard because m_state can be destroyed after releasing the guard
+                    expected = m_state.load(std::memory_order_relaxed);
```

Comment:
@BenFrantzDale, pay attention to the fix if you already started using the functionality (in worst case, the implementation without the fix might lead to memory corruptions or deadlocks).